x86 test failures due to comparing execution times with magic values. So these should be be disabled universally.

```sh
> FAILED tests/unit_tests/language_models/chat_models/test_rate_limiting.py::test_rate_limit_invoke - assert (1724100854.8413777 - 1724100854.6739) < 0.15
> FAILED tests/unit_tests/language_models/chat_models/test_rate_limiting.py::test_rate_limit_astream - assert 0.1 < (1724100856.3672216 - 1724100856.3302705)
```

--- a/pkgs/development/python-modules/langchain-core/default.nix
+++ b/pkgs/development/python-modules/langchain-core/default.nix
@@ -25,7 +25,7 @@
 
 buildPythonPackage rec {
   pname = "langchain-core";
-  version = "0.2.21";
+  version = "0.2.33";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -34,7 +34,7 @@ buildPythonPackage rec {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain-core==${version}";
-    hash = "sha256-8qEN03iimGLnhg6TdpPal+MXBZJ/QHJKwjxRF96abBw=";
+    hash = "sha256-vM3FY9E8PeC8LHP4QCTM1ggFynI+PscF7pv7CMaSZlU=";
   };
 
   sourceRoot = "${src.name}/libs/core";
@@ -52,10 +52,13 @@ buildPythonPackage rec {
     jsonpatch
     langsmith
     packaging
-    pydantic
     pyyaml
     tenacity
   ];
+
+  optional-dependencies = {
+    pydantic = [ pydantic ];
+  };
 
   pythonImportsCheck = [ "langchain_core" ];
 
@@ -86,15 +89,22 @@ buildPythonPackage rec {
     '';
   };
 
-  disabledTests = [
-    # flaky, sometimes fail to strip uuid from AIMessageChunk before comparing to test value
-    "test_map_stream"
-  ]
-  ++ lib.optionals stdenv.isDarwin [
-    # Langchain-core the following tests due to the test comparing execution time with magic values.
-    "test_queue_for_streaming_via_sync_call"
-    "test_same_event_loop"
-  ];
+  disabledTests =
+    [
+      # flaky, sometimes fail to strip uuid from AIMessageChunk before comparing to test value
+      "test_map_stream"
+      # Compares with machine-specific timings
+      "test_rate_limit_invoke"
+      "test_rate_limit_stream"
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      # Langchain-core the following tests due to the test comparing execution time with magic values.
+      "test_queue_for_streaming_via_sync_call"
+      "test_same_event_loop"
+      # Comparisons with magic numbers
+      "test_rate_limit_ainvoke"
+      "test_rate_limit_astream"
+    ];
 
   meta = {
     description = "Building applications with LLMs through composability";

--- a/pkgs/development/python-modules/langchain-openai/default.nix
+++ b/pkgs/development/python-modules/langchain-openai/default.nix
@@ -23,7 +23,7 @@
 
 buildPythonPackage rec {
   pname = "langchain-openai";
-  version = "0.1.17";
+  version = "0.1.22";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -32,7 +32,7 @@ buildPythonPackage rec {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain-openai==${version}";
-    hash = "sha256-ELD1KXCVx3SmiJodagtOHgBGKdjRWiRVCCNYcL63eCY=";
+    hash = "sha256-5UAijSTfQ6nQxdZvKHl2o01wDW6+Jphf38V+dAs7Ffk=";
   };
 
   sourceRoot = "${src.name}/libs/partners/openai";
@@ -48,7 +48,6 @@ buildPythonPackage rec {
   build-system = [ poetry-core ];
 
   dependencies = [
-    langchain
     langchain-core
     openai
     tiktoken

--- a/pkgs/development/python-modules/langchain/default.nix
+++ b/pkgs/development/python-modules/langchain/default.nix
@@ -31,7 +31,7 @@
 
 buildPythonPackage rec {
   pname = "langchain";
-  version = "0.2.9";
+  version = "0.2.14";
   pyproject = true;
 
   disabled = pythonOlder "3.8";
@@ -40,7 +40,7 @@ buildPythonPackage rec {
     owner = "langchain-ai";
     repo = "langchain";
     rev = "refs/tags/langchain==${version}";
-    hash = "sha256-HSr1watEDC28SZTsJWbXGrZsQd0O/wgxmj7pTsxMBOA=";
+    hash = "sha256-dgXcZu7dtmwlXp8dzHSNfbBnK7RWvrSwqYELm1fczzc=";
   };
 
   sourceRoot = "${src.name}/libs/langchain";
@@ -54,13 +54,16 @@ buildPythonPackage rec {
     langchain-core
     langchain-text-splitters
     langsmith
-    numpy
     pydantic
     pyyaml
     requests
     sqlalchemy
     tenacity
   ] ++ lib.optionals (pythonOlder "3.11") [ async-timeout ];
+
+  optional-dependencies = {
+    numpy = [ numpy ];
+  };
 
   nativeCheckInputs = [
     freezegun

